File: evals/refs/vowels.zig
const std = @import("std");

pub fn main(init: std.process.Init) !void {
    var in_buf: [4096]u8 = undefined;
    var fr = std.Io.File.stdin().reader(init.io, &in_buf);
    const r = &fr.interface;

    var out_buf: [4096]u8 = undefined;
    var fw = std.Io.File.stdout().writer(init.io, &out_buf);
    const w = &fw.interface;

    while (try r.takeDelimiter('\n')) |line| {
        var count: usize = 0;
        for (line) |c| {
            const lc = std.ascii.toLower(c);
            if (lc == 'a' or lc == 'e' or lc == 'i' or lc == 'o' or lc == 'u') {
                count += 1;
            }
        }
        try w.print("{d}\n", .{count});
    }
    try w.flush();
}

File: evals/refs/vowels.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func isVowel(c byte) bool {
	switch c | 0x20 {
	case 'a', 'e', 'i', 'o', 'u':
		return true
	}
	return false
}

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Buffer(make([]byte, 1024*1024), 1024*1024)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()
	for scanner.Scan() {
		line := scanner.Bytes()
		count := 0
		for _, c := range line {
			if isVowel(c) {
				count++
			}
		}
		fmt.Fprintln(writer, count)
	}
}
